src/utils/LicenseInfos.ts

```diff
@@ -0,0 +1,92 @@
+
+export function createLibraryLicenseInfos(pathToAssetsFolder: string): string {
+    let libraryLicenseInfos: LibraryLicenseInfo[] = JSON.parse(
+        fs
+            .readFileSync(
+                path.join(pathToAssetsFolder, LIBRARY_LICENSE_INFOS_NAME)
+            )
+            .toString()
+    );
+    libraryLicenseInfos.forEach((libraryLicenseInfo: LibraryLicenseInfo) => {
+        if (libraryLicenseInfo.licenseFile) {
+            const pathToInfos = path.join(
+                pathToAssetsFolder,
+                libraryLicenseInfo.licenseFile
+            );
+            if (fs.existsSync(pathToInfos)) {
+                libraryLicenseInfo.licenseText = fs
+                    .readFileSync(pathToInfos)
+                    .toString();
+            }
+        } else {
+            if (
+                !(
+                    libraryLicenseInfo.license
+                        .toLowerCase()
+                        .includes('commercial license') &&
+                    libraryLicenseInfo.license
+                        .toLowerCase()
+                        .includes('collaboration factory')
+                )
+            )
+                console.log(
+                    `Expected License File for ${libraryLicenseInfo.product} ${libraryLicenseInfo.component} does not exist ${libraryLicenseInfo.licenseFile}`
+                );
+        }
+    });
+
+    let licenseTexts = '';
+    libraryLicenseInfos.forEach((libraryLicenseInfo: LibraryLicenseInfo) => {
+        let includesLgpl = false;
+        if (libraryLicenseInfo.license.toLowerCase().includes('lgpl')) {
+            includesLgpl = true;
+        }
+        if (
+            libraryLicenseInfo.additionalLicenses &&
+            libraryLicenseInfo.additionalLicenses.length > 0
+        ) {
+            libraryLicenseInfo.additionalLicenses.forEach(
+                (additionalLicense) => {
+                    if (additionalLicense.toLowerCase().includes('lgpl')) {
+                        includesLgpl = true;
+                    }
+                }
+            );
+        }
+        if (includesLgpl) {
+            licenseTexts = licenseTexts.concat(ADDITIONAL_LGPL_TEXT + '\n');
+        }
+
+        licenseTexts = licenseTexts.concat(
+            'Component: ' +
```

Comment:
We will have to inline this info because of the performance. It should look like this...

```javascript
// [Component] lib-name-1, [Copyright] copyright-1., [License Text] GNU LESSER GENERAL PUBLIC LICENSE ...
// [Component] lib-name-2,, [Copyright] copyright-2, [License Text] GNU LESSER GENERAL PUBLIC LICENSE...

(()=>{var e={198:e=>{e.exports=function(e){function t(e){"undefined"!=typeof console&&(console.error||console.log)("[Script Loader]",e)...
```

Reply:
Text is now just prepended and it is not valid JS. Text will have to be JS comment in order to work. Check example above

![Screenshot 2023-12-12 at 08 47 32](https://github.com/collaborationFactory/cplace-asc/assets/16098085/ebb81a5b-4a37-4367-a022-03406f4e8aab)

Reply:
At the moment cplace-asc adds wrong license types to compressed.js. It should include GPL/LGPL, MPL, CC-BY-2.5

Reply:
is there a max-line length? and what is it?

Reply:
I would do it first without max-line length